src/material-table.js

```diff
@@ -440,7 +440,7 @@ export default class MaterialTable extends React.Component {
                 SelectProps={{
                   renderValue: value => <div style={{ padding: '0px 5px' }}>{value + ' ' + localization.labelRowsSelect + ' '}</div>
                 }}
-                page={this.isRemoteData() ? this.state.query.page : this.state.currentPage}
+                page={this.isRemoteData() ? (Math.min(this.state.query.page, this.state.query.totalCount)) : this.state.currentPage}
```

Comment:
This could be a bug. query.totalCount is not count of pages. It is count of total rows.